tests/ui/landing_page.py

```diff
@@ -143,7 +143,43 @@ def removeTestServer(self, name, server_url, should_raise_exception=True):
     # Wait for post to finish, can take a while.
     WebDriverWait(self.driver, UfOPageLayout.DEFAULT_TIMEOUT).until(
         EC.invisibility_of_element_located(((
-            LandingPage.SERVER_DELETE_SPINNER))))
+            LandingPage.SERVER_DETAILS_SPINNER))))
+
+  def editTestServer(self, old_name, ip, name, private_key, public_key,
+                     server_url):
+    """Edit a test server using the landing page (the only way).
+
+    Args:
+      old_name: A string for the name of the test server to edit.
+      ip: A string for the ip address of the server to insert.
+      name: A string for the name of the server to insert.
+      private_key: A string for the private key of the server to insert.
+      public_key: A string for the public key of the server to insert.
+      server_url: The base url portion of the landing page.
+    """
+    # Find the server and navigate to its details page.
+    self.driver.get(server_url + flask.url_for('landing'))
+    landing_page = LandingPage(self.driver)
+    server_list = landing_page.GetElement(LandingPage.SERVER_LIST_ITEM)
+    server_listbox = server_list.find_element(*LandingPage.GENERIC_LISTBOX)
+    server_item = landing_page.findItemInListing(server_listbox, old_name)
+
+    if server_item is None:
+      raise Exception
+    else:
+      server_item.click()
+
+    # Click edit on that server.
+    details_modal = server_item.find_element(*LandingPage.DETAILS_MODAL)
+    WebDriverWait(self.driver, UfOPageLayout.DEFAULT_TIMEOUT).until(
+        EC.visibility_of(details_modal))
+    edit_button = details_modal.find_element(*LandingPage.SERVER_EDIT_BUTTON)
+    edit_button.click()
+
+    # Perform the actual edits
+    add_edit_server_form = AddEditServerForm(self.driver)
+    add_edit_server_form.editTestServer(details_modal, ip, name, private_key,
```

Comment:
Just a couple of naming nits:

There is a couple of `editTestServer()` in different places, which would cause someone (me) to look at things closer to figure out how they are different.  I would recommend the one in the server form to just be called "editServer()".

Less important, do you think that we should just call `add_edit_server_form` to just be a more general `server_form`?
```python
from server_form import ServerForm
server_form = ServerForm(self.driver)
server_form.editTestServer(details_modal, ip, name, private_key, public_key)
```